### PR TITLE
Attach additional RHN Pools (post-provision custom action)

### PR DESCRIPTION
**Pull request by Tlacenka**
#### What does this PR do?
This PR adds a custom post-provision action playbook to the `custom-actions` folder and describes this as an example in the provisioning `README`

#### How should this be manually tested?
1) Run the provisioning using RHEL images.
2) Find out which other pools are available by running `subscription-manager list --available` command on one of the hosts.
3) Run `add-rhn-pools` to add these new pools like so:
```
ansible-playbook -i <inventory> openshift-ansible-contrib/playbooks/provisioning/openstack/custom-actions/add-rhn-pools.yml --extra-vars '{"rhn_pools":["<pool ID>","<pool ID>"]}' 
```
4) List available pools on one of the hosts by running `subscription-manager list --consumed` and confirm that the new pools have been added. 
#### Is there a relevant Issue open for this?
--

#### Who would you like to review this?
cc: @tomassedovic @bogdando @tzumainn PTAL

### CODE REVIEW

File: playbooks/provisioning/openstack/README.md [hunk 1]
```diff
@@ -535,13 +537,31 @@ This example runs against app nodes. The list of options include:
...
+---
+- hosts: cluster_hosts
+  tasks:
+  - name: Attach additional RHN pool
+    become: true
+    command: "/usr/bin/subscription-manager attach --pool=<pool ID>"
+    register: attach_rhn_pool_result
+    until: attach_rhn_pool_result.rc == 0
+    retries: 10
+    delay: 1
```
Comment (ioggstream) [inline playbooks/provisioning/openstack/README.md]: This delay may be too low for subscription-manager related issues.

**Comment by ioggstream** [comment]
Shouldn't we support multiple pools in here to be able to install OCP-on-OSP?

 https://github.com/openshift/openshift-ansible-contrib/blob/master/roles/subscription-manager/tasks/main.yml#L119

**Comment by patlachance** [comment]
How about handling RHN subscriptions by node type as in #749?

**Comment by Tlacenka** [comment]
@bogdando @ioggstream So if I understand it correctly, you want this to become a part of the `subscription-manager` role? No problem but it kind of defeats the purpose of it being just a custom post-provision action. Do we need this every time we run provisioning? ( @tomassedovic as the one who created this task, what do you think?)

File: playbooks/provisioning/openstack/README.md (hunk 1 above)
Comment (Tlacenka) [reply playbooks/provisioning/openstack/README.md]: what delay do you propose then?

Comment (ioggstream) [reply playbooks/provisioning/openstack/README.md]: afaik the cdn.redhat.com DNS record  TTL is 20 seconds. Maybe 10 or 15 seconds could be fine.

**Comment by ioggstream** [comment]
@Tlacenka if this is just a postinstall, we could:

  - implement the feature in the subscription_manager
  - reuse that role in this custom action, avoiding calling command

File: playbooks/provisioning/openstack/README.md (hunk 1 above)
Comment (Tlacenka) [reply playbooks/provisioning/openstack/README.md]: I see. I just used the same values as in the rest of `subscription-manager` tasks. I guess those should be adjusted as well then?

Comment (tomassedovic) [reply playbooks/provisioning/openstack/README.md]: @ioggstream this is a delay between retries not a timeout. If the task takes 30 seconds to succeed, this won't kill it.

It's just that if it fails, we will wait 1 second and retry. And do that 10 times.

**Comment by tomassedovic** [comment]
The intention of these sample playbooks is just to show people how to run some custom post-provision actions. It's mostly documentation that you happen to be able to run as a bonus.

If the deployer wants to do something these playbooks don't allow, they're expected to write their own or copy and update these.

As such, I am perfectly fine merging this.

I do get the point with with using the subscription-manager role. I see people reimplementing the same thing over and over in every red hat ansible repo. To my knowledge, other people in this repo with the CASL team are working on a more robust subscription manager role with the intention of upstreaming it either in openshift-ansible or maybe even ansible proper.

When that happens, we'll switch to it, but I don't want to block this little ad-hoc example on it.

**Pull request merged**